src/Ed25519VerificationKey2018.js

```diff
@@ -0,0 +1,439 @@
+/*!
+ * Copyright (c) 2018-2020 Digital Bazaar, Inc. All rights reserved.
+ */
+import * as bs58 from 'bs58';
+import * as env from './env.js';
+import * as forge from 'node-forge';
+import * as semver from 'semver';
+import * as util from './util.js';
+import * as _privateKeyNode12 from './ed25519PrivateKeyNode12.js';
+import * as _publicKeyNode12 from './ed25519PublicKeyNode12.js';
+import {createPublicKey, generateKeyPair, sign, verify} from 'crypto';
+import {LDVerifierKeyPair} from 'crypto-ld';
+import {promisify} from 'util';
+
+const {pki: {ed25519}, util: {binary: {base58}}} = forge;
+
+const SUITE_ID = 'Ed25519VerificationKey2018';
+
+class Ed25519VerificationKey2018 extends LDVerifierKeyPair {
+  /* eslint-disable max-len */
+  /**
+   * An implementation of
+   * [Ed25519 Signature 2018]{@link https://w3c-dvcg.github.io/lds-ed25519-2018/}
+   * for
+   * [jsonld-signatures.]{@link https://github.com/digitalbazaar/jsonld-signatures}
+   * @example
+   * > const privateKeyBase58 =
+   *   '3Mmk4UzTRJTEtxaKk61LxtgUxAa2Dg36jF6VogPtRiKvfpsQWKPCLesKSV182RMmvM'
+   *   + 'JKk6QErH3wgdHp8itkSSiF';
+   * > const options = {
+   *   publicKeyBase58: 'GycSSui454dpYRKiFdsQ5uaE8Gy3ac6dSMPcAoQsk8yq',
+   *   privateKeyBase58
+   * };
+   * > const EDKey = new Ed25519VerificationKey2018(options);
+   * > EDKey
+   * Ed25519VerificationKey2018 { ...
+   * @param {object} options - Options hashmap.
+   * @param {string} options.publicKeyBase58 - Base58 encoded Public Key
+   *   (unencoded is 32-bytes).
+   * @param {string} [options.privateKeyBase58] - Base58 Private Key
+   *   (unencoded is 64-bytes).
+   */
+  /* eslint-enable */
+  constructor(options = {}) {
+    super(options);
+    this.type = SUITE_ID;
+    this.publicKeyBase58 = options.publicKeyBase58;
+    if(!this.publicKeyBase58) {
+      throw new TypeError('The "publicKeyBase58" property is required.');
+    }
+    this.privateKeyBase58 = options.privateKeyBase58;
+    if(this.controller && !this.id) {
+      this.id = `${this.controller}#${this.fingerprint()}`;
+    }
+  }
+
+  /**
+   * Creates an instance of LDKeyPair from a key fingerprint.
+   * Note: Only key types that use their full public key in the fingerprint
+   * are supported (so, currently, only 'ed25519').
+   *
+   * @param {string} fingerprint
+   * @returns {LDKeyPair}
+   * @throws Unsupported Fingerprint Type.
+   */
+  static fromFingerprint({fingerprint}) {
```

Comment:
```suggestion
  static fromFingerprint({fingerprint} = {}) {
```

Reply:
We should probably also check to ensure `fingerprint` is a string.

Reply:
Fixed, now also checking if string.